src/controllers/wallet/p2sh/tx-proposal.controller.js

```diff
@@ -28,22 +27,30 @@ async function buildTxProposal(req, res) {
   const inputs = req.body.inputs || [];
   const changeAddress = req.body.change_address || null;
 
+  if (changeAddress && !await req.wallet.isAddressMine(changeAddress)) {
+    res.send({ success: false, error: 'Change address does not belong to the loaded wallet.' });
+    return;
+  }
+
   for (const output of outputs) {
     if (!output.token) {
       output.token = hathorLibConstants.HATHOR_TOKEN_CONFIG.uid;
     }
   }
   try {
-    // XXX: This is a temporary fix until the wallet-lib solves this storage issue
-    storage.setStore(req.wallet.store);
-    const sendTransaction = new SendTransaction({ outputs, inputs, changeAddress, network });
-    const tx = helpersUtils.createTxFromData(
-      { version: 1, ...sendTransaction.prepareTxData() },
-      network
-    );
+    const sendTransaction = new SendTransaction({
+      storage: req.wallet.storage,
+      outputs,
+      inputs,
+      changeAddress,
+    });
+    const txData = await sendTransaction.prepareTxData();
+    txData.version = 1;
+    const tx = transactionUtils.createTransactionFromData(txData, network);
 
     res.send({ success: true, txHex: tx.toHex() });
   } catch (err) {
+    console.error(err);
```

Comment:
Did you check the results from this? I usually get a `[object Object]` from this kind of log.

A `console.error(err.stack)` or `err.message` would probably be more helpful